tests/syntax/test_equals.py

```diff
@@ -0,0 +1,228 @@
+                   *[other] Other
+                }
+        """)
+
+        self.assertTrue(message1.equals(message1))
+
+    def test_same_complex_placeable_message(self):
+        message1 = self.parse_ftl_entry("""\
+            foo = Foo { NUMBER($num, style: "decimal") } Bar
+        """)
+
+        self.assertTrue(message1.equals(message1))
+
+    def test_same_message_with_attribute(self):
+        message1 = self.parse_ftl_entry("""\
+            foo
+                .attr = Attr
+        """)
+
+        self.assertTrue(message1.equals(message1))
+
+    def test_same_message_with_attributes(self):
+        message1 = self.parse_ftl_entry("""\
+            foo
+                .attr1 = Attr 1
+                .attr2 = Attr 2
+        """)
+
+        self.assertTrue(message1.equals(message1))
+
+    def test_same_message_with_tag(self):
+        message1 = self.parse_ftl_entry("""\
+            foo = Foo
+                #tag
+        """)
+
+        self.assertTrue(message1.equals(message1))
+
+    def test_same_message_with_tags(self):
+        message1 = self.parse_ftl_entry("""\
+            foo = Foo
+                #tag1
+                #tag2
+        """)
+
+        self.assertTrue(message1.equals(message1))
+
+    def test_same_junk(self):
+        message1 = self.parse_ftl_entry("""\
+            foo = Foo {
+        """)
+
+        self.assertTrue(message1.equals(message1))
+
+
+class TestOrderEquals(unittest.TestCase):
+    def setUp(self):
+        self.parser = FluentParser()
+
+    def parse_ftl_entry(self, string):
+        return self.parser.parse_entry(dedent_ftl(string))
+
+    def test_attributes(self):
+        message1 = self.parse_ftl_entry("""\
+            foo
+                .attr1 = Attr1
+                .attr2 = Attr2
+        """)
+        message2 = self.parse_ftl_entry("""\
+            foo
+                .attr2 = Attr2
+                .attr1 = Attr1
+        """)
+
+        self.assertTrue(message1.equals(message2))
+        self.assertTrue(message2.equals(message1))
+
+    def test_tags(self):
+        message1 = self.parse_ftl_entry("""\
+            foo = Foo
+                #tag1
+                #tag2
+        """)
+        message2 = self.parse_ftl_entry("""\
+            foo = Foo
+                #tag2
+                #tag1
+        """)
+
+        self.assertTrue(message1.equals(message2))
+        self.assertTrue(message2.equals(message1))
+
+    def test_variants(self):
+        message1 = self.parse_ftl_entry("""\
+            foo =
+                { $num ->
+                    [a] A
+                   *[b] B
+                }
+        """)
+        message2 = self.parse_ftl_entry("""\
+            foo =
+                { $num ->
+                   *[b] B
+                    [a] A
+                }
+        """)
+
+        self.assertTrue(message1.equals(message2))
+        self.assertTrue(message2.equals(message1))
+
+
+class TestSpansEqual(unittest.TestCase):
```

Comment:
I guess the name here is older than your inequality test? Rename this to TestWithSpans or so?